### ISSUE TITLE
Google client config token is not returned when creating a second GKE cluster in the same project

### ISSUE THREAD

**Issue by Stringls** [terraform-google-modules/terraform-google-kubernetes-engine#1725]
### TL;DR

When I am trying to create a second GKE cluster in the same project, different/same project and zone, using another TF cloud workspace I am able to create infrastructure in GCP, but when it comes to provisioning Kubernetes resources it fails with an error `User "system:anonymous" cannot create resource` that makes me thinking it due to an empty access token. Please check the Terraform configuration.

### Expected behavior

When I create a second GKE cluster access token from Google cloud config is returned and Kubernetes resources are provisioned.

### Observed behavior

The infrastructure (a second GKE cluster, network, NAT etc.) are provisioned, but access token from Google cloud config is not returned so Kubernetes resources, Helm releases are not provisioned.

### Terraform Configuration

```hcl
data "terraform_remote_state" "state" {
  backend = "remote"

  config = {
    organization = ""
    workspaces = {
      name = ""
    }
  }
}

data "google_client_config" "default" {}

provider "kubernetes" {
  host                   = "https://${module.gke.endpoint}"
  token                  = data.google_client_config.default.access_token
  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
}

terraform {
  required_providers {

    google = {
      source  = "hashicorp/google"
      version = "~> 4.0"
    }
    helm = {
      source  = "hashicorp/helm"
      version = "2.9.0"
    }
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = "2.20.0"
    }
  }
  cloud {
    organization = ""

    workspaces {
      name = ""
    }
  }
}

provider "google" {
  project = var.project_id
  region  = var.region
}

provider "google-beta" {
  project = var.project_id
  region  = var.region
}

provider "helm" {
  kubernetes {
    host                   = "https://${module.gke.endpoint}"
    token                  = data.google_client_config.default.access_token
    cluster_ca_certificate = base64decode(module.gke.ca_certificate)
  }
}

module "gke" {
  source                            = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  project_id                        = var.project_id
  name                              = var.gke_cluster_name 
  region                            = var.region
  zones                             = [var.zone]
  network                           = module.gke-network.network_name
  subnetwork                        = module.gke-network.subnets_names[0]
  ip_range_pods                     = module.gke-network.subnets_secondary_ranges[0].*.range_name[0]
  ip_range_services                 = module.gke-network.subnets_secondary_ranges[0].*.range_name[1]
  create_service_account            = true
  remove_default_node_pool          = true
  description                       = ""
  enable_shielded_nodes             = false
  http_load_balancing               = false
  network_policy                    = false
  enable_private_endpoint           = false
  enable_private_nodes              = true
  master_ipv4_cidr_block            = "172.16.0.16/28"
  horizontal_pod_autoscaling        = false
  filestore_csi_driver              = false
  add_master_webhook_firewall_rules = true
  firewall_inbound_ports            = ["443", "6443", "8443", "9443", "15017", "9440", "10250", "10259", "10257", "2379", "2380"]
  enable_identity_service           = true
  #  kubernetes_version         = latest by default

  node_pools = [
    {
      name               = var.node_pool
      machine_type       = var.machine_type
      autoscaling        = false
      initial_node_count = 1
      image_type         = "COS_CONTAINERD"
      disk_size_gb       = 100
      disk_type          = "pd-standard"
      enable_gcfs        = false
      enable_gvnic       = false
      auto_repair        = true
      auto_upgrade       = true
      preemptible        = false
      node_count         = var.node_count 
    },
  ]

  node_pools_oauth_scopes = {
    all = [
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/trace.append",
      "https://www.googleapis.com/auth/service.management.readonly",
      "https://www.googleapis.com/auth/monitoring",
      "https://www.googleapis.com/auth/devstorage.read_only",
      "https://www.googleapis.com/auth/servicecontrol",
    ]
    mgmt-node-pool = [
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/trace.append",
      "https://www.googleapis.com/auth/service.management.readonly",
      "https://www.googleapis.com/auth/monitoring",
      "https://www.googleapis.com/auth/devstorage.read_only",
      "https://www.googleapis.com/auth/servicecontrol",
    ]
  }

  node_pools_labels = {
    all = {}

    mgmt-node-pool = {

    }
  }

  node_pools_metadata = {
    all = {}

    mgmt-node-pool = {}
  }


  node_pools_tags = {
    all = []

    mgmt-node-pool = []
  }
}

module "gke-network" {
  source  = "terraform-google-modules/network/google"
  version = ">= 4.0.1"

  project_id   = var.project_id
  network_name = var.network_name

  subnets = [
    {
      subnet_name   = var.subnet_name
      subnet_ip     = var.subnet_ip_range
      subnet_region = var.region
    },
  ]

  firewall_rules = var.firewall_rules

  secondary_ranges = {
    (var.network_name) = [
      {
        range_name    = "cidr-pods"
        ip_cidr_range = "10.1.0.0/16"
      },
      {
        range_name    = "cidr-services"
        ip_cidr_range = "10.2.0.0/20"
      },
  ] }

}

module "cloud-nat" {
  source        = "terraform-google-modules/cloud-nat/google"
  project_id    = var.project_id
  region        = var.region
  create_router = true
  name          = var.cloud-nat-name
  network       = module.gke-network.network_name
  router        = var.cloud-router-name
}

resource "kubernetes_namespace" "vault_secrets_operator" {
  metadata {
    name = var.vault_secrets_operator_namespace
  }
}

variable "project_id" {
  description = "The GCP project to use for integration tests"
  type        = string
  default     = ""
}

variable "region" {
  description = "The GCP region to create and test resources in"
  type        = string
  default     = "us-central1"
}

variable "zone" {
  description = "The GCP zone to create and test resources in"
  type        = string
  default     = "us-central1-a"
}

variable "cloud-nat-name" {
  description = "The GCP cloud nat name to create and test resources in"
  type        = string
  default     = "test-gateway"
}

variable "cloud-router-name" {
  description = "The GCP cloud router name to create and test resources in"
  type        = string
  default     = "test-router"
}

variable "network_name" {
  description = "The name of the network create this instance in."
  type        = string
  default     = "test-network"
}

variable "subnet_name" {
  description = "The name of the subnetwork create this instance in."
  type        = string
  default     = "test-subnet"
}

variable "subnet_ip_range" {
  description = "Range of IP adresses in subnet"
  default     = "10.0.0.0/24"
  type        = string
}

variable "gke_cluster_name" {
  description = "The name of the node pool"
  type        = string
  default     = "test-cluster"
}

variable "node_pool" {
  description = "The name of the node pool"
  type        = string
  default     = "test-node-pool"
}


variable "node_count" {
  description = "The name of the node pool"
  type        = number 
  default     = 3 
}

variable "machine_type" {
  description = "The type of the machine that will be created"
  type        = string
  default     = "e2-custom-2-4096"
}

variable "firewall_rules" {
  description = "Firewall rules for VMs"
  default = [
    {
      name                 = "test-allow-ssh"
      direction            = "INGRESS"
      ranges               = ["0.0.0.0/0"]
      use_service_accounts = false
      targets              = null
      priority             = 1001
      sources              = null
      allow = [
        {
          protocol = "tcp"
          ports    = ["22"]
        }
      ]
      deny = []
    }
  ]

  type = list(any)
}

variable "vault_secrets_operator_namespace" {
  description = "The namespace of Vault Secrets operator"
  type        = string
  default     = "vault-secrets-operator-system"
}
```

### Terraform Version

```sh
Terraform v1.5.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.80.0
+ provider registry.terraform.io/hashicorp/google-beta v4.80.0
+ provider registry.terraform.io/hashicorp/helm v2.9.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.20.0
+ provider registry.terraform.io/hashicorp/random v3.5.1
```

### Additional information

When I create a first cluster, it does not happend. The token is returned, it's present in the Terraform state. It happens only when I try to create a second GKE cluster in the same project, using (I am not actually sure as I am not able to check creds in workspace for the first cluster) the same credentials. The infra is provisioned, so Provider is authenticated into API. I've also checked this issue https://github.com/hashicorp/terraform-provider-google/issues/14602, but the upgrading to the latest version didn't help me.

**Comment by Stringls** [comment]
Any updates on that ^?

**Comment by Stringls** [comment]
Any updates on that ^?
